cmd/object-handlers.go

```diff
@@ -583,6 +583,10 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 
 	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex)
 	if err != nil {
+		if _, ok := err.(hex.InvalidByteError); ok {
+			writeErrorResponse(w, ErrContentSHA256Mismatch, r.URL)
```

Comment:
we have a `pkg/hash/errors.go`  to return this error SHA256Mismatch{} @krisis we so we can return from the hash.NewReader() internally we don't need to do this here. 

We can perhaps do this for md5sum as well while we are at it.

Reply:
You can also bring a test case inside the package pkg/hash for this.